### DISCUSSION TITLE
openDTU 23.12.16 keine verbindung mit dem HMS-2000-4T moeglich

### DISCUSSION THREAD

**Discussion by snchlore** [tbnobody/OpenDTU#2041] [Q&A]
Hallo zusammen,

mit der openDTU [v23.11.16](https://github.com/tbnobody/OpenDTU/releases/tag/v23.11.16) funktioniert die Verbindung zum Wechselrichter HMS-2000-4T einwandfrei.

Wenn ich auf  [v23.12.16](https://github.com/tbnobody/OpenDTU/releases/tag/v23.12.16) umsteige, kann keine Verbindung mehr aufgebaut werden und ich soll die Kabelverbindungen überprüfen.

Hat jemand eine Idee?

Hier sind die Daten des Wechselrichters:

 Wechselrichter-Informationen | |
-------- | --------
Seriennummer | ***********
Produktionsjahr | 2022
Produktionswoche | 1
Modell | HMS-2000
Ermittelte max. Leistung | 2.000 W
Bootloader-Version | 0.0.1
Firmware-Version | 1.0.12
Firmware-Erstellungsdatum | 2021-10-25 18:37:00
Hardware-Teilenummer | 269644033
Hardware-Version | 01.00

Firmwareinformationen | |
-------- | --------
Hostname	|OpenDTU-62F868
SDK-Version	|v4.4.4
Konfigurationsversion	|0.1.25
Firmwareversion / git Hash	|[v23.11.16](https://github.com/tbnobody/OpenDTU/releases/tag/v23.11.16)
PIO Umgebung	|generic
Firmware-Aktualisierung	|[Neue Version verfügbar! Zeige Änderungen!](https://github.com/tbnobody/OpenDTU/compare/v23.11.16...HEAD)
Reset Grund CPU 0	|Vbat power on reset
Reset Grund CPU 1	|for APP CPU, reset by PRO CPU
Anzahl der Konfigurationsspeicherungen	|9
Betriebszeit	|0 days 00:09:36

Hardwareinformationen | |
-------- | --------
Chip-Modell	|ESP32-D0WD-V3
Chip-Revision	|3
Chip-Kerne	|2
CPU-Frequenz	|240 MHz

Funkmodulinformationen | |
-------- | --------
nRF24 Status  |nicht konfiguriert
nRF24 Chip-Status	||
nRF24 Chip-Type	|unbekannt
CMT2300A Status	|konfiguriert
CMT2300A Chip-Status	|verbunden

Kategorie | Name | Ausgewählt | Aktiv
-------- | -------- | -------- | --------
Cmt | clk | 12 | 12
|| cs | 27 | 27 
|| fcs | 26	| 26
|| gpio2 | -1	| -1
|| gpio3	|-1	|-1
|| sdio	|14	|14
Display	|clk	|22	|22
 ||cs	|	|255
 ||data	|21	|21
 ||reset	|	|255
 ||type	|3	|3
Eth	|clk_mode	|	|0
 ||enabled	|	|false
 ||mdc		||23
 ||mdio		||18
 ||phy_addr		||0
 ||power		||-1
 ||type		||0
Led	|led0	|25|	25
 ||led1	|26	|26
Nrf24	|clk	|-1|	-1
 ||cs	|-1	|-1
 ||en	|-1	|-1
 ||irq	|-1	|-1
 ||miso	|-1	|-1
 ||mosi	|-1	|-1

**Comment by ghost** [comment]


**Reply by snchlore** [reply]
Besten Dank
Habe zuerst den esp komplett gelöscht, dann die [generic_esp32.factory (23.11.16)](https://github.com/tbnobody/OpenDTU/releases/download/v23.11.16/opendtu-generic_esp32.factory.bin) aufgespielt, config.bin und pin.bin hochgeladen. Funktioniert einwandfrei.

Habe die esp wieder komplett gelöscht, dann die [generic_esp32.factory (24.6.10)](https://github.com/tbnobody/OpenDTU/releases/download/v24.6.10/opendtu-generic_esp32.factory.bin) aufgespielt, config.bin und pin.bin hochgeladen. Hat kurz funktioniert.
Jetzt kommt wieder der Fehler "Es konnte keine Verbindung zu einem der konfigurierten Funkmodule hergestellt werden. Bitte überprüfen Sie die Verdrahtung."

**Comment by snchlore** [comment]
Wenn ich die Config.bin neu aufspiele und den Inverter neu hinzufuege, funktioniert es fuer eine kurze Zeit von ca. 43s

```
17:50:55.126 > Fetch inverter: ***********
17:50:55.162 > TX RealTimeRunData 865.00 MHz --> 15 80 15 28 75 80 18 46 28 80 0B 00 66 69 C3 DF 00 00 00 00 00 00 00 00 D9 D0 BA 
17:50:55.222 > Interrupt received
17:50:55.266 > RX 865.00 MHz --> 95 80 15 28 75 80 18 46 28 01 00 01 01 A6 01 A6 00 5D 00 5E 01 89 01 90 00 06 B7 | -35 dBm
17:50:55.286 > Interrupt received
17:50:55.295 > RX 865.00 MHz --> 95 80 15 28 75 80 18 46 28 02 76 AC 00 01 5F 2E 03 E6 03 EF 00 05 00 05 00 00 0A | -35 dBm
17:50:55.322 > Interrupt received
17:50:55.332 > RX 865.00 MHz --> 95 80 15 28 75 80 18 46 28 03 00 00 00 00 00 00 00 04 EE 7A 00 00 00 00 00 00 38 | -35 dBm
17:50:55.369 > Interrupt received
17:50:55.383 > RX 865.00 MHz --> 95 80 15 28 75 80 18 46 28 04 00 00 09 27 13 86 02 F1 00 00 00 20 03 E7 00 CA E9 | -35 dBm
17:50:55.409 > Interrupt received
17:50:55.417 > RX 865.00 MHz --> 95 80 15 28 75 80 18 46 28 85 00 05 A9 DF 5D | -35 dBm
17:50:55.742 > RX Period End
17:50:55.742 > Success
17:50:55.945 > TX AlarmData 865.00 MHz --> 15 80 15 28 75 80 18 46 28 80 11 00 66 69 C3 DF 00 00 00 00 00 00 00 00 03 CB 61 
17:50:56.070 > Interrupt received
17:50:56.077 > RX 865.00 MHz --> 95 80 15 28 75 80 18 46 28 01 00 01 80 01 00 01 28 3A 28 3A 00 00 00 00 00 D3 F8 | -35 dBm
17:50:56.110 > Interrupt received
17:50:56.118 > RX 865.00 MHz --> 95 80 15 28 75 80 18 46 28 02 00 02 28 41 00 00 00 00 00 00 00 D4 00 03 28 41 7C | -35 dBm
17:50:56.150 > Interrupt received
17:50:56.168 > RX 865.00 MHz --> 95 80 15 28 75 80 18 46 28 03 00 00 00 00 00 00 00 DC 00 04 29 6D 00 00 00 00 34 | -35 dBm
17:50:56.191 > Interrupt received
17:50:56.202 > RX 865.00 MHz --> 95 80 15 28 75 80 18 46 28 84 00 DC 00 DE 00 05 29 6D 00 00 00 00 00 DC 3B D3 58 | -35 dBm
17:50:56.562 > RX Period End
17:50:56.563 > Success
17:50:59.563 > Websocket: [/livedata][1] connect
17:51:00.151 > Fetch inverter: ************
17:51:00.151 > Request device info
17:51:00.160 > TX RealTimeRunData 865.00 MHz --> 15 80 15 28 75 80 18 46 28 80 0B 00 66 69 C3 E4 00 00 00 00 00 00 00 00 E8 F7 97 
.
.
.
.
.
17:51:37.311 > TX SPI Timeout
17:51:38.134 > RX Period End
17:51:38.134 > All missing
17:51:38.134 > Nothing received, resend whole request
17:51:38.134 > TX AlarmData 860.00 MHz --> 15 80 15 28 75 80 18 46 28 80 11 00 66 69 C4 05 00 00 00 00 00 00 00 00 BC 99 51 
```

**Reply by snchlore** [reply]
Danke, danke, danke!!!

Das sieht gut aus.
